src/app/common/components/MustGatherWatcher.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { Button } from '@patternfly/react-core';
+import { AppContext } from '@app/common/context';
+import { useMustGatherQuery } from '@app/queries';
+import { mustGatherStatus } from '@app/client/types';
+import { Link } from 'react-router-dom';
+
+interface IMustGatherWatcherProps {
+  name: string;
+  listStatus: mustGatherStatus;
+}
+
+export const MustGatherWatcher: React.FunctionComponent<IMustGatherWatcherProps> = ({
+  name,
+  listStatus,
+}: IMustGatherWatcherProps) => {
+  const { pushNotification } = React.useContext(AppContext);
+  const completedPreviously = listStatus === 'completed';
```

Comment:
Trying to figure out if these variables could have better names, since the behavior isn't obvious right away... I think maybe they're as good as they'll get. I was thinking of like `completedOnMount` / `erroredOnMount` but those might be more confusing, not less. Maybe a comment could help, but it's also fine as is.

This watcher pattern is so slick, man. Nice work.

Reply:
Yea naming these was tricky.. happy to update it if you can think of something more descriptive. Might be good enough for a first pass..